### PR TITLE
MediaCollectionViewCell: Added a basic MediaCell and structure for ad…

### PR DESCRIPTION
**Pull request by carolanitz**
…ding more cells

This also removes the old VLCPlaylistTableViewCell and VLCPlaylistCollectionViewCell

### Checklist
- [x] I've run `bundle exec fastlane test` from the root directory to see all new and existing tests pass
- [x] I've followed the [vlc-ios code style](Docs/CodingStyle.md) and run `bundle exec fastlane lint` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/videolan/vlc-ios#contribute)
- [x] I've updated the documentation if necessary.

### Description

 Issues that still need to be addressed:
-  [ ] The cell is not dynamic in height, it will be the same height for titles with one line and titles with two lines
-  [ ] The cells don't respect the notch 
-  [ ] The cells don't update yet when the media updates. This is because the code doesn't update media at this point when playback is done and the file was played

### CODE REVIEW

File: SharedSources/MovieCollectionViewCell.swift [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    @IBOutlet weak var descriptionLabel: UILabel!
+    @IBOutlet weak var progressView: UIProgressView!
+
+    override class var nibName: String {
+        return "MovieCollectionViewCell"
+    }
+
+    override var media: VLCMLObject? {
+        didSet {
+            guard let movie = media as? VLCMLMedia else {
```
Comment (carolanitz) [inline SharedSources/MovieCollectionViewCell.swift]: all this code should move into a function

File: SharedSources/MovieCollectionViewCell.swift [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        backgroundColor = PresentationTheme.current.colors.background
+        titleLabel?.textColor = PresentationTheme.current.colors.cellTextColor
+        descriptionLabel?.textColor = PresentationTheme.current.colors.cellDetailTextColor
+    }
+
+    override func prepareForReuse() {
+        super.prepareForReuse()
+        titleLabel.text = ""
+        descriptionLabel.text = ""
+        thumbnailView.image = nil
```
Comment (carolanitz) [inline SharedSources/MovieCollectionViewCell.swift]: progress isn't handled here and the newLabel

File: SharedSources/MediaLibraryModel/MediaLibraryBaseModel.swift [hunk 3]
```diff
@@ -44,6 +45,10 @@ protocol MLBaseModel: MediaLibraryBaseModel {
 }
 
 extension MLBaseModel {
+    var cellType: MediaCollectionViewCell.Type {
+        return MovieCollectionViewCell.self
```
Comment (Mikanbu) [inline SharedSources/MediaLibraryModel/MediaLibraryBaseModel.swift]: I'm not sure to understand this addition, what happens will with the GenreModel for example?

File: Sources/MediaCategories/MediaCategoryViewController.swift [hunk 4]
```diff
@@ -224,7 +223,8 @@ class VLCMediaCategoryViewController: UICollectionViewController, UICollectionVi
         var cellWidth = collectionView.bounds.size.width / numberOfCells
         cellWidth = cellWidth - ceil(((numberOfCells + 1) * cellPadding) / numberOfCells)
 
-        return CGSize(width: cellWidth, height: cellWidth * aspectRatio)
+        // 3*20 for the labels + 24 for 3*8 which is the padding
+        return CGSize(width: cellWidth, height: cellWidth * aspectRatio + 3*20+24)
```
Comment (carolanitz) [inline Sources/MediaCategories/MediaCategoryViewController.swift]: this part will be adjusted with a dynamic calculation

File: SharedSources/MediaLibraryModel/VideoModel.swift [hunk 5]
```diff
@@ -95,12 +95,25 @@ extension VLCMLMedia {
...
     }
 
     @objc func formatSize() -> String {
         return ByteCountFormatter.string(fromByteCount: Int64(mainFile().size()),
                                          countStyle: .file)
     }
+
+    func mediaProgress() -> Float {
+        guard let string = metadata(of: .progress).str as NSString? else {
+            return 0.0
```
Comment (Mikanbu) [inline SharedSources/MediaLibraryModel/VideoModel.swift]: Is this for safety?

File: Sources/VLCPlaybackController.m [hunk 6]
```diff
@@ -375,9 +375,7 @@ - (void)_savePlaybackState
         APLog(@"failed to save current media state - file removed?");
     }
 }
-#endif
 
-#if TARGET_OS_IOS
```
Comment (Mikanbu) [inline Sources/VLCPlaybackController.m]: I don't understand this VLCPlaybackController change in this commit.
I believe it should be a separate commit.

File: SharedSources/MovieCollectionViewCell.swift [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+class MovieCollectionViewCell: MediaCollectionViewCell {
+
+    @IBOutlet weak var thumbnailView: UIImageView!
+    @IBOutlet weak var titleLabel: UILabel!
+    @IBOutlet weak var newLabel: UILabel!
+    @IBOutlet weak var descriptionLabel: UILabel!
+    @IBOutlet weak var progressView: UIProgressView!
+
+    override class var nibName: String {
+        return "MovieCollectionViewCell"
```
Comment (Mikanbu) [inline SharedSources/MovieCollectionViewCell.swift]: What about using `String(describing:)` here?

File: SharedSources/MediaLibraryModel/MediaLibraryBaseModel.swift (hunk 3 above)
Comment (carolanitz) [reply SharedSources/MediaLibraryModel/MediaLibraryBaseModel.swift]: It's a fallback for now until this is implemented

File: SharedSources/MediaLibraryModel/VideoModel.swift (hunk 5 above)
Comment (carolanitz) [reply SharedSources/MediaLibraryModel/VideoModel.swift]: when there is no progress yet set the default will be 0

File: Sources/VLCPlaybackController.m (hunk 6 above)
Comment (carolanitz) [reply Sources/VLCPlaybackController.m]: could be. I just saw it and removed it

File: SharedSources/MovieCollectionViewCell.swift (hunk 7 above)
Comment (carolanitz) [reply SharedSources/MovieCollectionViewCell.swift]: could use it yes

File: SharedSources/MediaLibraryModel/MediaLibraryBaseModel.swift (hunk 3 above)
Comment (carolanitz) [reply SharedSources/MediaLibraryModel/MediaLibraryBaseModel.swift]: exactly :)

Comment (Mikanbu) [reply SharedSources/MediaLibraryModel/MediaLibraryBaseModel.swift]: Since it won't always be a media, I think we should we consider renaming it so that it can be more generic.

What about`BaseCollectionViewCell`?

Comment (carolanitz) [reply SharedSources/MediaLibraryModel/MediaLibraryBaseModel.swift]: oh interesting so I can't always set a media on the cell? so I guess the base type is a VLCMLFile ?

Comment (Mikanbu) [reply SharedSources/MediaLibraryModel/MediaLibraryBaseModel.swift]: >  I can't always set a media on the cell?

Not sure to understand what you mean here, for example, you cannot have `VLCMLMedia` in a genre category.

The base class of the VLCMediaLibraryKit models would be `VLCMLObject`.
`VLCMLFile` is different from the old MediaLibrary, it is files contained in a certain Media.

**Comment by Mikanbu** [comment]
Merged with d5225e9b522d22c0f057606c619c90ae5677fdb1.
